pkg/analyzer/analyze.go

```diff
@@ -0,0 +1,665 @@
+    return nil, fmt.Errorf("failed to located package based on purl")
+  }
+  return pkgResponse, nil
+}
+
+func FindHasSBOMBy(filter model.HasSBOMSpec, uri, purl, id string, ctx context.Context, gqlclient graphql.Client) (*model.HasSBOMsResponse, error) {
+  var foundHasSBOMPkg *model.HasSBOMsResponse
+  var err error
+  if purl != "" {
+    pkgResponse, err := getPkgResponseFromPurl(ctx, gqlclient, purl)
+    if err != nil {
+      fmt.Printf("getPkgResponseFromPurl - error: %v", err)
+      return nil, err
+    }
+    foundHasSBOMPkg, err = model.HasSBOMs(ctx, gqlclient, model.HasSBOMSpec{Subject: &model.PackageOrArtifactSpec{Package: &model.PkgSpec{Id: &pkgResponse.Packages[0].Namespaces[0].Names[0].Versions[0].Id}},
+      })
+    if err != nil {
+      fmt.Printf("(purl)failed getting hasSBOM with error :%v", err)
+      return nil, err
+    }
+  } else if uri != ""{
+    foundHasSBOMPkg, err = model.HasSBOMs(ctx, gqlclient, model.HasSBOMSpec{Uri: &uri,})
+    if err != nil {
+      fmt.Printf("(uri)failed getting hasSBOM  with error: %v", err)
+      return nil, err
+    }
+  }else if id != ""{
+    foundHasSBOMPkg, err = model.HasSBOMs(ctx, gqlclient, model.HasSBOMSpec{Id: &id,})
+    if err != nil {
+      fmt.Printf("(id)failed getting hasSBOM  with error: %v", err)
+      return nil, err
+    }
+  } else {
+    foundHasSBOMPkg, err = model.HasSBOMs(ctx, gqlclient, filter)
+    if err != nil {
+      fmt.Printf("(filter)failed getting hasSBOM  with error: %v", err)
+      return nil, err
+    }
+  }
+  return foundHasSBOMPkg, nil
+}
+
+func verifyAnalyzeFlags(slsas, sboms []string,  errSlsa, errSbom error, uri, purl, id bool) {
+
+  if (errSlsa != nil && errSbom != nil) || (len(slsas) ==0  && len(sboms) == 0 ){
+    fmt.Println("Must specify slsa or sboms ")
+    os.Exit(0)
+  }
+
+  if len(slsas) >0  && len(sboms) >0 {
+    fmt.Println("Must either specify slsa or sbom")
+    os.Exit(0)
+  }
+
+  if errSlsa == nil && (len(slsas) <= 1|| len(slsas) > 2) && len(sboms) == 0{
+    fmt.Println("Must specify exactly two slsas to analyze, specified", len(slsas))
+    os.Exit(0)
+  }
+
+  if errSbom == nil && (len(sboms) <= 1|| len(sboms) > 2) && len(slsas) == 0{
+    fmt.Println("Must specify exactly two sboms to analyze, specified", len(sboms))
+    os.Exit(0)
+  }
+
+  if errSlsa == nil && len(slsas) == 2 {
+    fmt.Println("slsa diff to be implemented.")
+    os.Exit(0)
+  }
+
+  if !uri && !purl && !id {
+    fmt.Println("Must provide one of --uri or --purl")
+    os.Exit(0)
+  }
+
+  if uri && purl  || uri && id || purl && id {
+    fmt.Println("Must provide only one of --uri or --purl")
+    os.Exit(0)
+  }
+}
+
+
+func GenerateAnalysisOutput(analysisGraph graph.Graph[string, *Node], diffList HighlightedDiff, all, dot bool, maxprint, action int, gqlclient graphql.Client){
+  //Create dot file
+  createGraphDotFile(dot, analysisGraph)
+  //print to stdout
+  printHighlightedAnalysis(dot, diffList, all, maxprint, action, analysisGraph )
```

Comment:
Anything under `pkg/` should not print anything to stdout. Any logic should result in a datastructure that returns back to `cmd/` and is then printed. This helps for testability. Looks like much of this may need to be moved back to `cmd/`